crates/bevy_render/src/camera/camera.rs

```diff
@@ -67,9 +68,9 @@ pub fn camera_system<T: CameraProjection + Component>(
         changed_window_ids.push(event.id);
     }
 
-    for (mut camera, mut camera_projection) in &mut query.iter() {
+    for (entity, mut camera, mut camera_projection) in &mut query.iter() {
         if let Some(window) = windows.get(camera.window) {
-            if changed_window_ids.contains(&window.id) {
+            if changed_window_ids.contains(&window.id) || query_added.entity(entity).is_ok() {
```

Comment:
While this does "work" in that it ensures the camera gets updated, It doesn't work as expected because afaik it will run every frame. This happens for two reasons:

1) `query.entity()` returns a QueryOne (if the query matches the entity archetype), it doesn't actually run the query on the given entity.
2. even if we were to run `query_one.get()` (to run the query) it wouldn't actually take into account the `Added<T>` filter. This is a bug in QueryOne as far as I'm concerned. Inside `query_one.get()` we also need to run `.should_skip()` before calling `.next()`

Would you be open to fixing that problem too? It would amount to returning None here if `fetch.should_skip()` is true: https://github.com/bevyengine/bevy/blob/afe86b4b3a5bfd3a9a7f71375f4ad25296e6ba68/crates/bevy_ecs/hecs/src/query_one.rs#L47

Reply:
Yes, I’ll look into that also. Thanks for the catch !